src/time_integrator.f90

```diff
@@ -8,10 +8,16 @@ module m_time_integrator
   type :: time_intg_t
     integer :: istep, nsteps, nsubsteps, order, nvars, nolds
     type(flist_t), allocatable :: olds(:, :)
+    type(flist_t), allocatable :: curr(:)
+    type(flist_t), allocatable :: deriv(:)
```

Comment:
Using flist_t to generalize time-integration schemes to an arbitrary number of variables